docs/user/guides/setup/configuration-file.rst

```diff
@@ -0,0 +1,223 @@
+How to add a configuration file
+===============================
+
+As part of the initial setup for your Read the Docs site,
+you need to create a **configuration file** called ``.readthedocs.yaml``.
+The configuration file tells Read the Docs what specific settings to use for your project.
+
+This how-to guide covers:
+
+#. Where to put your configuration file.
+#. Templates to setup your configuration file from scratch.
+#. The most basic sections in the configuration file that can help you get started.
+#. Next steps...
+
+.. seealso::
+
+   :doc:`/tutorial/index`.
+     Following the steps in our tutorial will help you setup your first documentation project.
+
+
+Where to put your configuration file
+------------------------------------
+
+The ``.readthedocs.yaml`` file should be placed in the top-most directory of your project's repository.
+We will get to the contents of the file in the next steps.
+
+When you have changed the configuration file,
+you need to commit and push the changes to your Git repository.
+Read the Docs will then automatically find and use the configuration to build your project.
+
+.. note::
+
+    Why is it called .yaml? What is that?
+      The configuration file is a `YAML`_ file. YAML files are a "map": a collection of
+      key-value pairs that can be nested. This is not unlike a JSON file or ``dict``
+      object in Python. This page won't explain the structure of YAML files, but many resources exist
+      online.
+
+
+.. _YAML: https://en.wikipedia.org/wiki/YAML
```

Comment:
This section felt too low level for our docs. I think it's good to call out the fact that this is a YAML file, but we shouldn't need to explain YAML as a concept. If the user isn't familiar, they likely aren't familiar with most of the concepts/terms in this block anyways -- map, key-value paris, JSON, dict, etc.

To retain describing the structure without getting too deep into technical jargon, I think something close to this is best:

```suggestion
.. note::

    The Read the Docs configuration file is a `YAML`_ file. You can read more on the structure of this file and all of our configuration options :doc:`in our configuration file reference <>`.
```

Reply:
I felt like this was a strong point in @nikblanchet's original howto: Cluing in the user so they don't feel like the guide is over their head if they haven't seen YAML before. You are not expected to have previous experience with YAML in order to follow this guide and copy the examples.

That's why the headline was: "Why is it called .yaml? What is that?" - to meet the user at their level.

I'm not sure about the change here: Do you still want to keep the reference to the Wikipedia article? I added that _again_ because I think it's great to flag that: No, you do not need to have met the concept of YAML before reading this guide, but here is a Wikipedia article explaining it.

Reply:
If the note is just gonna be a cross-reference to the configuration file reference, I think we should delete it completely, already plenty of those :)

Reply:
I do agree with the point about trying to be inclusive with this page, but I also feel familiarity with YAML is part of our base user persona. So, the copy/voice felt like it wasn't talking to a true user, to talking past our normal user -- it would feel the same as having a section explaining what Linux or what a terminal is to me.

I do agree on the cross reference, it isn't horribly important. I added it there to make the note more meaningful and pad out the content. I just updated my suggestion to remove this and link out to better resources. I meant to find a better link than Wikipedia, but agree that linking out is a good resource for this block.

For some comparison, here is the tutorial from CircleCI:
https://circleci.com/docs/config-intro/

It doesn't mention YAML in any detail, but does link out to a separate doc just for explaining YAML in more detail:
https://circleci.com/docs/introduction-to-yaml-configurations/

This is maybe a good addition for later for our docs, though I might argue we'd have trouble maintaining this if we're _just now_ adding a tutorial on our own configuration file :smile: 

From this page, and also from the YAML spec website, YAML is described as:

> YAML is a human-friendly data serialization language for all programming languages.

That's a good sweet spot for our description :dart:

I've updated the suggestion with this and linked to the YAML website. There is a link to the YAML spec, _but_ the overview and introduction are approachable and not frighteningly technical like specs normally are.